rust/Makefile

```diff
@@ -59,11 +60,56 @@ alloc-cfgs = \
     --cfg no_rc \
     --cfg no_sync
 
+proc_macro2-skip_flags := \
+    --edition=2021 \
+    -Drust_2018_idioms \
+    -Dunreachable_pub \
+    -Dunsafe_op_in_unsafe_fn
```

Comment:
Maybe use `--cap-lints allow` instead like cargo does for non-local dependencies?

Reply:
rustc is very loathe to add warnings, but one of the larger categories we do add to often enough is future incompat warnings. Those eventually upgrade to errors, given enough time, if they are that bad. So the "code quality" may be just fine today but have some subtle detail that recommends a different pattern in the future, and in that case, you will eventually find the problem when it actually becomes upgraded.

Reply:
In actual practice, I expect the kernel will use only well-tested, fully-developed but slightly-active projects that have their CI run about once every kernel release, or at worst once every few. So the most stringent I would recommend is running a cron job at either rustc's 6 week or the kernel's ~10 week cadence to check for new developments in dependencies.

Reply:
> Maybe use --cap-lints allow instead like cargo does for non-local dependencies?

Yeah, good idea, we should do this for dependencies that we have not modified (or at least very minimally for build purposes), i.e. like these, but unlike `alloc` for the moment.